### ISSUE TITLE
Can't redefine schema when map type field nested in multiple levels

### ISSUE THREAD

**Issue by dejank1986** [ImFlog/schema-registry-plugin#126]
Starting from version `1.9.0` we have the following issue when running `testSchemasTask`:
```
Could not parse Avro schema
org.apache.avro.SchemaParseException: Can't redefine: com.test.seller.EntitlementDetails
```

Looks like there is some issue when having map type field nested in multiple levels. 
The event definition:
```
{
  "type": "record",
  "name": "SellerUpdated",
  "namespace": "com.test.seller.event",
  "fields": [
    {
      "name": "entitlements",
      "type": {
        "type": "map",
        "values": {
          "type": "map",
          "values": "com.test.seller.FeatureEntitlements"
        }
      }
    },
    {
      "name": "id",
      "type": "long"
    }
  ]
}
```

FeatureEntitlements local reference:
```
{
  "name": "FeatureEntitlements",
  "type": "record",
  "namespace": "com.test.seller",
  "fields": [
    {
      "name": "global",
      "type": [
        "null",
        "com.test.seller.EntitlementDetails"
      ]
    },
    {
      "name": "listings",
      "type": {
        "type": "map",
        "values": "com.test.seller.EntitlementDetails"
      }
    }
  ]
}
```

EntitlementDetails local reference:
```
{
  "name": "EntitlementDetails",
  "type": "record",
  "namespace": "com.test.seller",
  "fields": [
    {
      "name": "maxItems",
      "type": [
        "null",
        "int"
      ]
    },
    {
      "name": "remainingItems",
      "type": [
        "null",
        "int"
      ]
    }
  ]
}
```

Also looks like the last schema version published to Schema Registry was also not generated properly. If you check values in the `listings` field, it's not complex type like it should be, since `EntitlementDetails` is record type:
```
{
  "type": "record",
  "name": "SellerUpdated",
  "namespace": "com.test.event",
  "fields": [
    {
      "name": "entitlements",
      "type": {
        "type": "map",
        "values": {
          "type": "map",
          "values": {
            "type": "record",
            "name": "FeatureEntitlements",
            "namespace": "com.test.entitlements",
            "fields": [
              {
                "name": "global",
                "type": [
                  "null",
                  {
                    "type": "record",
                    "name": "EntitlementDetails",
                    "fields": [
                      {
                        "name": "maxItems",
                        "type": [
                          "null",
                          "int"
                        ]
                      },
                      {
                        "name": "remainingItems",
                        "type": [
                          "null",
                          "int"
                        ]
                      }
                    ]
                  }
                ]
              },
              {
                "name": "listings",
                "type": {
                  "type": "map",
                  "values": "EntitlementDetails"
                }
              }
            ]
          }
        }
      }
    },
    {
      "name": "id",
      "type": "long"
    }
  ]
}
```

Not sure it's exactly related to the map type field, but generated schema definition does not look correct.

**Comment by ImFlog** [comment]
Hello @dejank1986, thank you for your intested in this plugin.

It indeed looks like a bug, can you share the gradle configuration linked to the error (the schema registry part) ?

From what I understand in the error message this may be due to the fact that we inline the record where it is referenced and thus if we try to do it twice, it fails -> `can't redefine: com.test.seller.EntitlementDetails`.
I need to check what could be done in Avro to bypass this limitation.

I will probably have a bit of time next week to look more into this issue, but If you are willing to help on this, a reproducing test would be a great start and the local reference replacement is in the `AvroSchemaParser` file.

Thank you

**Comment by dejank1986** [comment]
Gradle config:
```
def subjects = [
   [subject: 'seller-event-com.test.seller.event.SellerUpdated', schema: 'src/main/avro/seller/seller-updated.avsc', 
      dependencies: [
         [name: 'com.test.seller.FeatureEntitlements', path: 'src/main/avro/seller/feature-entitlements.avsc'],
         [name: 'com.test.seller.EntitlementDetails', path: 'src/main/avro/seller/entitlement-details.avsc']
      ]
   ]
]

compatibility {
        subjects.each {
            s ->
                def subject = subject(s.subject, s.schema)
                s.dependencies.each {
                    d -> subject.addLocalReference(d.name, d.path)
                }
        }
    }
```

Steps to reproduce:
- Use plugin version 1.8.0
- Create avro schema SellerUpdated with local references (provided in first comment)
- Publish first schema version
- Update plugin version to 1.9.0
- Test compatibility (should fail with error from first comment)

**Comment by dejank1986** [comment]
Hi @ImFlog, any update on this issue? I see it's still present even after upgrading plugin to version `1.11.0`.

**Comment by ImFlog** [comment]
The version 1.11.0 only brings a new feature.
I am close to merge this PR #132 that should help for your issue.
Will keep you posted when done to let you test with the new version.
Sorry for the troubles.

**Comment by dejank1986** [comment]
Thanks @ImFlog for the update. Looking forward to a solution.

**Comment by ImFlog** [comment]
Hello @dejank1986, I've just released the version 1.11.1. Can you try with this version to see if the issue was related ?
NB: It requires to add Jitpack in the build repository (see README) because we are now using a custom Avro version to allow us to handle correctly local + remote dependency mixing in Avro.
Thanks

**Comment by dejank1986** [comment]
Hi @ImFlog,

I've just gave it a try and it works well. Thanks a lot of addressing this issue.
